sheets/2/exercise-6.tex

```diff
@@ -0,0 +1,74 @@
+\excercise{Wenn ich mich doch nur nach links drehen könnte}
+
+\begin{enumerate}[label=\alph*)]
+    \item Instanziiere den Task und den Verifier. Dabei soll wie in der \textit{Aufgabe 5b)} der SubTask jeweils angegeben werden.
+    \item Lasse Neo im Uhrzeigersinn an der Wand entlang laufen, bis er wieder an seinem Startpunkt angekommen ist.
+    Implementiere das Verhalten in der Methode \textit{subTaskB()} in der Klasse \textit{Sheet3Task6}.
+    \item Lasse nun Neo gegen den Uhrzeigersinn an der Wand entlang laufen, bis er wieder an seinem Startpunkt angekommen ist.
+    Implementiere das Verhalten in der Methode \textit{subTaskC()} in der Klasse \textit{Sheet3Task6}.
+    Was fällt dir im Vergleich zu vorhin auf?
+\end{enumerate}
+\begin{Infobox}
+    Wir wollen uns nun die Methodensyntex etwas genauer anschauen. Dafür betrachten wir beispielhaft den folgenden Programmcode:
+    \begin{lstlisting}[breaklines=true, numbers=none]
+//Class A
+public void foo(){
+    B b = new B();
+    System.out.println(b.bar(5, 7));
+}
+
+//Class B
+private int b = 9;
+public boolean bar(int a, int b){
+    if(a > b)
+        return true;
+    return power(a) > this.b;
+}
+
+private int power(int a){
+    return a*a;
+}
+\end{lstlisting}
+Bei der Programmausführung gehen wir davon aus dass die Methode \textit{foo} ausgeführt wird. Wir unterteilen die Erklärung der Syntax in folgende Blöcke:
```

Comment:
Am besten nochmal in der Syntax Aufgabe von Blatt 1 nachsehen, was schon erklärt wurde.

Reply:
hier ist der Schnitt dann doch relativ gering. Zur Vollständigkeit hat man dann im Notfall vielleicht ein paar Infos drin, die den Studis schon bekannt sind.

Reply:
Ja dann lieber vollständig erklären.